### PR TITLE
Fix reconfiguration end-to-end test

### PR DESCRIPTION
**Pull request by jumaffre**
The `reconfiguration` end-to-end test had become very flaky recently and this PR fixes the instability. 

The issue was that for the reconfiguration test, a node with a untrusted code version was trying to join the network, triggering its quote verification by the leader. Because the quote verification currently takes ~2 secs and the join timer was around the same period (see `nodestate.h`), the leader's node ring buffer was getting filled with expensive join requests, either not processing other join requests or client connections.

The join timer has now been relaxed in our tests to 4 seconds (configurable via `cchost`, default to 1 second) and our python client waits for a joining node to show up in the leader's store for twice that duration.

### CODE REVIEW

File: tests/reconfiguration.py [hunk 1]
```diff
@@ -86,10 +86,10 @@ def run(args):
     ) as network:
         network.start_and_join(args)
         test_add_node_from_backup(network, args)
-        test_add_as_many_pending_nodes(network, args)
         test_add_node(network, args)
         test_add_node_untrusted_code(network, args)
         test_retire_node(network, args)
+        test_add_as_many_pending_nodes(network, args)
```
Comment (jumaffre) [inline tests/reconfiguration.py]: Moved this to the end of the test run as this would subsequently spam the leader with join requests.

**Comment by cimetrics** [comment]
fix_reconfiguration@[3731 aka 20200108.8](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3731&view=results) vs master ewma over 20 builds from [3612](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3612&view=results) to [3727](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3727&view=results)
images

File: src/host/main.cpp [hunk 2]
```diff
@@ -235,6 +235,14 @@ int main(int argc, char** argv)
       true)
     ->check(CLI::ExistingFile);
 
+  size_t join_timer = 1000;
+  join->add_option(
+    "--join-timer",
+    join_timer,
+    "Frequency at which the joining node will try to join an existing network "
```
Comment (eddyashton) [inline src/host/main.cpp]: Minor: This is a delay (ms), not a frequency. Would prefer something like "Time after which the joining node will resend join requests to existing network (ms)".

File: tests/infra/ccf.py [hunk 3]
```diff
@@ -238,7 +239,10 @@ def create_and_add_pending_node(self, lib_name, host, args, target_node=None):
             self.consortium.wait_for_node_to_exist_in_store(
                 primary,
                 new_node.node_id,
-                (
+                # When the service is open, it takes a joining node at least 2 join
+                # attempts to retrieve the network secrets
+                timeout=int(args.join_timer * 3 / 1000),
```
Comment (eddyashton) [inline tests/infra/ccf.py]: Why `int`? Assuming this is eventually just passed to `time.sleep` or equivalent, a float is fine. If we're rounding to whole seconds for a reason, we probably want to round up with `math.ceil` (join_timer == 999ms => wait 3 seconds).

**Comment by cimetrics** [comment]
fix_reconfiguration@[3736 aka 20200108.11](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3736&view=results) vs master ewma over 20 builds from [3612](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3612&view=results) to [3727](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3727&view=results)
images

File: tests/infra/ccf.py (hunk 3 above)
Comment (jumaffre) [reply tests/infra/ccf.py]: We actually pass this value to a loop (`for _ in range(timeout):`) hence the integer value. `math.ceil()` is better though 👍

**Comment by cimetrics** [comment]
fix_reconfiguration@[3738 aka 20200108.12](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3738&view=results) vs master ewma over 20 builds from [3612](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3612&view=results) to [3727](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3727&view=results)
images

**Pull request merged**